sky/packages/sky/lib/src/rendering/editable_paragraph.dart

```diff
@@ -0,0 +1,115 @@
+// Copyright 2015 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:sky/painting.dart';
+import 'package:sky/src/rendering/box.dart';
+import 'package:sky/src/rendering/object.dart';
+import 'package:sky/src/rendering/paragraph.dart';
+import 'package:sky/src/rendering/proxy_box.dart' show SizeChangedCallback;
+
+const _kCursorGap = 1.0;
+const _kCursorHeightOffset = 2.0;
+const _kCursorWidth = 1.0;
+
+/// A render object used by EditableText widgets.  This is similar to
+/// RenderParagraph but also renders a cursor and provides support for
+/// scrolling.
+class RenderEditableParagraph extends RenderParagraph {
+
```

Comment:
If you have two newlines after the {, put two newlines before the } at the end of the class.